tests/interface/TWIoTeXStakingTests.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Data.h"
+#include "HexCoding.h"
+#include "PrivateKey.h"
+#include "IoTeX/Signer.h"
+#include "IoTeX/Staking.h"
+#include "proto/IoTeX.pb.h"
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::IoTeX;
+
+TEST(TWIoTeXSigner, Staking) {
+    auto input = Proto::SigningInput();
+    input.set_version(1);
+    input.set_nonce(123);
+    input.set_gaslimit(888);
+    input.set_gasprice("999");
+    auto keyhex = parse_hex("0806c458b262edd333a191e92f561aff338211ee3e18ab315a074a2d82aa343f");
+    input.set_privatekey(keyhex.data(), keyhex.size());
+
+    // staking is implemented using the Execution message
+    auto staking = input.mutable_execution();
+    staking->set_amount("456");
+    staking->set_contract(STAKING_CONTRACT);
+    // call staking API to generate calldata
+    byte canName[12] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+    auto data = Stake(canName, 1001, true, 0, nullptr);
```

Comment:
There is no way in Swift or Java to call methods in `Staking.cpp`, you have two possible solutions:

0. Ethereum-way, implement something like `TWEthereumAbiEncoder` and let iOS / Android app to encode the contract data
1. Tron-way, define new message in proto file and encode the data in cpp

Reply:
i see, thanks for the explanation, will modify accordingly

Reply:
please check latest 1 commit